neqo-crypto/src/agent.rs

```diff
@@ -97,7 +97,8 @@ macro_rules! preinfo_arg {
 
 impl SecretAgentPreInfo {
     fn new(fd: *mut ssl::PRFileDesc) -> Res<Self> {
-        let mut info: ssl::SSLPreliminaryChannelInfo = unsafe { mem::uninitialized() };
+        let mut info: ssl::SSLPreliminaryChannelInfo =
```

Comment:
This is not the correct usage of `MaybeUninit`. You should write

```rust
let mut info: MaybeUninit<ssl::SSLPreliminaryChannelInfo> =
    MaybeUninit::uninit();

secstatus_to_res(unsafe {
    ssl::SSL_GetPreliminaryChannelInfo(
        fd,
        info.as_mut_ptr(),
        c_uint::try_from(mem::size_of::<ssl::SSLPreliminaryChannelInfo>())?,
    )
})?;

Ok(Self {
    info: info.assume_init(),
    alpn: get_alpn(fd, true)?,
})

Reply:
I *thought* that was the way to use it, but the Rust documentation "told" me otherwise. See here: https://doc.rust-lang.org/beta/std/mem/union.MaybeUninit.html. That said, I understand that the documentation is making a point about instantaneously triggering the undefined behavior. So, of course, you are correct! Sending in a fix.

Reply:
Looks like we also need an unsafe{} around anywhere that we use assume_init(). Does this make sense to you?

Reply:
Yes, `assume_init `requires unsafe. I have neglected this.